soc/xtensa/intel_adsp/common/ace/asm_memory_management.h

```diff
@@ -0,0 +1,76 @@
+/* Copyright (c) 2022 Intel Corporation
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @brief Macros for power gating memory banks specific for ACE 1.0
+ */
+
+#ifndef __Z_ACE_LIB_ASM_MEMORY_MANAGEMENT_H__
+#define __Z_ACE_LIB_ASM_MEMORY_MANAGEMENT_H__
+
+#ifdef _ASMLANGUAGE
+
+/* These definitions should be placed elsewhere, but I can't find a good place for them. */
+#define LSPGCTL			  0x71D80
+#define MAX_MEMORY_SEGMENTS       1
+#define EBB_SEGMENT_SIZE	  32
+#define PLATFORM_HPSRAM_EBB_COUNT 22
+
+.macro m_ace_hpsram_power_change segment_index, mask, ax, ay, az, au, aw
```

Comment:
@andyross those macros disable lp/hp sram. And since we disable memory we want ace_power_down to be a single function that is executed before shut down of a core.

IMR is slow and we don't wanna be slow.